### PR TITLE
[sailfish-browser] Remove default search engine guard. Fixes JB#32815

### PR DESCRIPTION
**Pull request by rainemak**
It's enough to check that search engines has been initialized.

@pvuorela / @jpetrell quick one to review ?

### CODE REVIEW

File: src/core/settingmanager.cpp [hunk 1]
```diff
@@ -199,9 +199,8 @@ void SettingManager::handleObserve(const QString &message, const QVariant &data)
                 mozContext->sendObserve(QLatin1String("embedui:search"), QVariant(removeMsg));
             }
 
-            // After first start we can update the default search engine immediately
-            // without
-            if (m_searchEnginesInitialized && defaultSearchEngine.isEmpty()) {
+            // After first start we can update the default search engine immediately.
+            if (m_searchEnginesInitialized) {
                 setSearchEngine();
```
Comment (siteshwar) [inline src/core/settingmanager.cpp]: ```setSearchEngine()``` method checks if ```m_searchEnginesInitialized``` is set. Probably this check can be removed altogether.

Comment (rainemak) [inline src/core/settingmanager.cpp]: As, true.

**Comment by rainemak** [comment]
@siteshwar updated.

**Comment by siteshwar** [comment]
LGTM

**Pull request merged**